Description: define PATH_MAX if undefined
 We're (ab)using PATH_MAX as an upper-bound on a string buffer size,
 but it isn't defined on all platforms (notably HURD). For those
 platforms just defined a reasonably high number.
Forwarded: via email (ignored)
Author: Jonathan Dowland <jmtd@debian.org>
Last-Update: 2014-05-22

diff --git a/tty.c b/tty.c
index ac9ed44..1b83016 100644
--- a/tty.c
+++ b/tty.c
@@ -35,6 +35,10 @@
 #include "main.h"
 #include "mincurses.h"
 
+#ifndef PATH_MAX
+#  define PATH_MAX 4096
+#endif
+
 static int force=0;
 static int p_tm,p_ch,p_b,p_r,p_fm,p_rm,pcm_n,p_tr,p_tmax,p_pau,p_pl,p_tn,p_g;
 static double p_st,p_cur,p_end,p_len;

Description: Fix X/X/Y test logic
Forwarded: via email (ignored)
Bug-Debian: http://bugs.debian.org/753776
Author: Thibaut Girka <thib@sitedethib.com>
Reviewed-By: Jonathan Dowland <jmtd@debian.org>

--- a/main.c
+++ b/main.c
@@ -222,7 +222,7 @@
       r[1] = 1-r[1];
       r[2] = 1-r[2];
     }
-    *cchoice = (r[0]==r[1] ? 2 : (r[1]==r[2] ? 0 : 1));
+    *cchoice = r[(r[0]==r[1] ? 2 : (r[1]==r[2] ? 0 : 1))];
     break;
   }
 }
